File: gaia-sdk-java/gaia-sdk-graphql/src/main/kotlin/gaia/sdk/response/type/ApiKey.kt
package gaia.sdk.response.type

import gaia.sdk.Uuid
import gaia.sdk.ISO8601
import gaia.sdk.Struct
import gaia.sdk.response.intf.*
import com.fasterxml.jackson.annotation.JsonProperty
import com.fasterxml.jackson.annotation.JsonCreator
import gaia.sdk.request.enumeration.*

/**
* Represents api key information
*/
data class ApiKey @JsonCreator constructor(
    /**
    * The api key id
    */
    @JsonProperty("apiKeyId") val apiKeyId:Uuid? = null, 
    /**
    * The name of the api key
    */
    @JsonProperty("name") val name:String? = null, 
    /**
    * The description of the api key
    */
    @JsonProperty("description") val description:String?, 
    /**
    * The secret of the api key
    */
    @JsonProperty("secret") val secret:String? = null, 
    /**
    * The flag to enable the api key
    */
    @JsonProperty("enabled") val enabled:Boolean? = null
)

File: gaia-sdk-python/gaia_sdk/graphql/response/type/ApiKey.py
Uuid = str
String = str
ISO8601 = str
Struct = dict
Float = float
Int = int
Boolean = bool


class ApiKey:
    """
    Represents api key information
    """
    dictionary: dict

    def __init__(self, dictionary: dict):
        self.dictionary = dictionary

    def __eq__(self, other):
        if type(other) is type(self):
            return self.dictionary == other.dictionary
        return False

    def __repr__(self):
        return {'dictionary': self.dictionary}

    """
    The api key id
    """
    @property
    def api_key_id(self) -> Uuid:
        return Uuid(self.dictionary.get("apiKeyId"))
    """
    The name of the api key
    """
    @property
    def name(self) -> String:
        return String(self.dictionary.get("name"))
    """
    The description of the api key
    """
    @property
    def description(self) -> String:
        return String(self.dictionary.get("description"))
    """
    The flag to enable the api key
    """
    @property
    def enabled(self) -> Boolean:
        return Boolean(self.dictionary.get("enabled"))
